src/classes/ALLO_AllocationsService_TEST.cls

```diff
@@ -0,0 +1,853 @@
+
+        // Opportunity 10 - Existing allocations.  Resize for Opportunity Amount Difference
+        message = baseMessage + ' - Opportunity 10';
+        allocs = allocWrapper.oppsAllocations.get(oppIds[10]);
+        ALLO_UnitTestHelper_TEST.assertSObjectList(allocs, 2, message);
+        alloc0 = ALLO_UnitTestHelper_TEST.findByGAU(allocs, gaus[1]);
+        alloc1 = ALLO_UnitTestHelper_TEST.findByGAU(allocs, defaultGAU);
+        ALLO_UnitTestHelper_TEST.assertOpportunityAllocation(alloc0, oppIds[10], 10, 50, gaus[1].Id, message + ' - Allocation 0');
+        ALLO_UnitTestHelper_TEST.assertOpportunityAllocation(alloc1, oppIds[10], 10, null, defaultGAU.Id, message + ' - Allocation 1');
+
+        // Opportunity 11 - Invalid existing fixed allocation that is too large for the Opportunity Amount
+        message = baseMessage + ' - Opportunity 11';
+        allocs = allocWrapper.oppsAllocations.get(oppIds[11]);
+        ALLO_UnitTestHelper_TEST.assertSObjectList(allocs, 1, message);
+        alloc0 = ALLO_UnitTestHelper_TEST.findByGAU(allocs, gaus[1]);
+        ALLO_UnitTestHelper_TEST.assertOpportunityAllocation(alloc0, oppIds[11], 40, null, gaus[1].Id, message + ' - Allocation 0');
+        System.assertEquals(true, errorsByRecordId.containsKey(oppIds[11]), message + ' - Error Record Present');
+
+        // Recurring Donation - Allocations
+        message = baseMessage + ' - Recurring Donation';
+        rdAllocs = allocWrapper.recurringDonationAllocations.get(rdId);
+        ALLO_UnitTestHelper_TEST.assertSObjectList(rdAllocs, 2, message);
+        alloc0 = ALLO_UnitTestHelper_TEST.findByGAU(rdAllocs, gaus[0]);
+        alloc1 = ALLO_UnitTestHelper_TEST.findByGAU(rdAllocs, gaus[1]);
+        ALLO_UnitTestHelper_TEST.assertRecurringDonationAllocation(alloc0, rdId, null, 50, gaus[0].Id, message + ' - Allocation 0');
+        ALLO_UnitTestHelper_TEST.assertRecurringDonationAllocation(alloc1, rdId, 10, null, gaus[1].Id, message + ' - Allocation 1');
+
+        // Campaign - Allocations
+        message = baseMessage + ' - Campaign';
+        campAllocs = allocWrapper.campaignAllocations.get(campaignId);
+        ALLO_UnitTestHelper_TEST.assertSObjectList(campAllocs, 1, message);
+        alloc0 = ALLO_UnitTestHelper_TEST.findByGAU(campAllocs, gaus[1]);
+        ALLO_UnitTestHelper_TEST.assertCampaignAllocation(alloc0, campaignId, 10, null, gaus[1].Id, message + ' - Allocation 0');
+
+        if (alloSetupTest.isPaymentAllocationsEnabled()) {
+            assertPaymentAllocationsAfterProcessing(allocWrapper.paymentAllocations, oppIds, gaus, defaultGAU);
+        }
+    }
+
+    /**
+     * @description Executes various Opportunity and Payment Allocation Scenarios when
+     *              Payment and Default Allocations are not
+     *              [ALLO_AllocationsService.processRecords]
+     */
+    @isTest
+    private static void confirmAllocationsServiceUpdatesAllocationsCorrectlyWhenDefaultAllocationsAreNotEnabled() {
```

Comment:
This method is another huge individual test method that looks a lot like the large method above. Can these be further abstracted to use common helper methods and/or (better) into smaller tests that test individual functionality?

Reply:
See above